--- syslog_process.php	2016-05-17 13:53:10.049972728 +0800
+++ /usr/share/cacti/site/plugins/syslog/syslog_process.php	2016-05-17 15:12:58.001932644 +0800
@@ -505,33 +505,37 @@
 $syslog_reports = sizeof($reports);
 
 syslog_debug("We have " . $syslog_reports . " Reports in the database");
-
 if (sizeof($reports)) {
 foreach($reports as $syslog_report) {
+	// 20160517 modified by klting
 	print '   Report: ' . $syslog_report['name'] . "\n";
+	$syslog_report['min'] = ($syslog_report['timepart']%3600)/60; 
+ 	$syslog_report['hour'] = floor($syslog_report['timepart']/3600); 
 	if ($syslog_report['min'] < 10)
 		$syslog_report['min'] = '0' . $syslog_report['min'];
 
-	$base_start_time = $syslog_report['hour'] . ' : ' . $syslog_report['min'];
-
+	$date = new DateTime(); 
+	$date->setTimestamp((floor($syslog_report['date']/86400))*86400); 
+	$base_start_time = $date->format('Y-m-d') . " " .$syslog_report['hour'] . ':' . $syslog_report['min'] .":00"; 
+	
 	$current_time = strtotime("now");
 	if (empty($last_run_time)) {
 		if ($current_time > strtotime($base_start_time)) {
 			/* if timer expired within a polling interval, then poll */
 			if (($current_time - 300) < strtotime($base_start_time)) {
-				$next_run_time = strtotime(date("Y-m-d") . " " . $base_start_time);
+				$next_run_time = strtotime($base_start_time);
 			}else{
-				$next_run_time = strtotime(date("Y-m-d") . " " . $base_start_time) + 3600*24;
+				$next_run_time = strtotime($base_start_time) + $syslog_report['timespan'];
 			}
 		}else{
-			$next_run_time = strtotime(date("Y-m-d") . " " . $base_start_time);
+			$next_run_time = strtotime($base_start_time);
 		}
 	}else{
 		$next_run_time = $last_run_time + $seconds_offset;
 	}
 	$time_till_next_run = $next_run_time - $current_time;
 
-	if ($next_run_time < 0) {
+	if ($time_till_next_run < 0) {
 		print '       Next Send: Now' . "\n";
 		print "       Creating Report...\n";
 
